src/dparse/parser.d

```diff
@@ -5288,8 +5294,27 @@ class Parser
             }
             if (currentIs(tok!")"))
                 break;
+
+            // Save starting point to deal with attributed variadics, e.g.
+            // int printf(in char* format, scope const ...);
+            const startIdx = index;
+            auto cp = allocator.setCheckpoint();
+
             if (!parameters.put(parseParameter()))
-                return null;
+            {
+                // parseParameter fails for C-style variadics, they are parsed below
+                if (!currentIs(tok!"..."))
+                    return null;
+
+                // Reset to the beginning of the current parameters
+                index = startIdx;
+                allocator.rollback(cp);
+
+                node.hasVarargs = true;
+                mixin(parseNodeQ!(`node.varargsAttributes`, `VariadicArgumentsAttributes`));
+                expect(tok!"...");
```

Comment:
I think that this should be `if (expect(tok!"...") is null) return null;`, as the parser functions should return null if there is an error. I know that not a lot of the other parse functions work this way right now, but that's a but that needs to be fixed some other time.

Reply:
Fixed